OsmAnd/src/net/osmand/plus/plugins/CustomOsmandPlugin.java

```diff
@@ -83,6 +93,77 @@ public CustomOsmandPlugin(@NonNull OsmandApplication app, @NonNull JSONObject js
 		readAdditionalDataFromJson(json);
 		readDependentFilesFromJson(json);
 		loadResources();
+		updateCustomRegionsState();
+	}
+
+	private void updateCustomRegionsState() {
+		ArrayList<Pair<File, File>> filesToCopy = new ArrayList();
+		for (WorldRegion region : getFlatCustomRegions()) {
+			if (region instanceof CustomRegion) {
+				CustomRegion customRegion = (CustomRegion) region;
+				List<IndexItem> indexItems = customRegion.loadIndexItems();
+				for (IndexItem item : indexItems) {
+					if (item.isHidden()) {
+						item.setIsHidden(false);
+						File nonHiddenFile = item.getTargetFile(app);
+						if (nonHiddenFile.exists()) {
+							item.setIsHidden(true);
+							File hiddenFile = item.getTargetFile(app);
+							filesToCopy.add(new Pair<>(nonHiddenFile, hiddenFile));
+						}
+						File nonHiddenBackupedFile = new File(app.getAppPath(BACKUP_INDEX_DIR), nonHiddenFile.getName());
+						if (nonHiddenBackupedFile.exists()) {
+							File hiddenBackupedFile = new File(app.getAppInternalPath(HIDDEN_BACKUP_DIR), nonHiddenFile.getName());
+							filesToCopy.add(new Pair<>(nonHiddenBackupedFile, hiddenBackupedFile));
+						}
+					}
+				}
+			}
+		}
+		CopyHiddenFilesTask task = new CopyHiddenFilesTask(filesToCopy);
+		task.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR);
+	}
+
+	class CopyHiddenFilesTask extends AsyncTask<Void, Void, Void> {
+		ArrayList<Pair<File, File>> filesToCopy;
+
+		CopyHiddenFilesTask(ArrayList<Pair<File, File>> filesToCopy) {
+			this.filesToCopy = filesToCopy;
+		}
+
+		@Override
+		protected Void doInBackground(Void... voids) {
+			for (Pair<File, File> fileToCopy : filesToCopy) {
+				File origin = fileToCopy.first;
+				File target = fileToCopy.second;
+				boolean success = true;
+				File parent = target.getParentFile();
+				if (parent != null && !parent.exists()) {
+					parent.mkdirs();
+				}
+				try (InputStream in = new FileInputStream(origin)) {
+					try (OutputStream out = new FileOutputStream(target)) {
+						byte[] buf = new byte[1024];
+						int len;
+						while ((len = in.read(buf)) > 0) {
```

Comment:
Use Algorithms.fileCopy() or Algorithms.streamCopy() if custom copy is not necessary